mqtt.go

```diff
@@ -298,6 +299,16 @@ func (a *applicationPubSub) AllDevices() DeviceSub {
 	return a.Device("+")
 }
 
+func (a *applicationPubSub) Publish(devID string, downlink *types.DownlinkMessage) error {
+	d := &devicePubSub{
+		logger: a.logger,
+		client: a.client,
+		appID:  a.appID,
+		devID:  devID,
```

Comment:
I tried to be consistent with

```
Publish(devID string, downlink *types.DownlinkMessage) error
Device(devID string) DevicePubSub
```

otherwise I think it was going to be a bit "magic"

Of course we can also just optimize the `Device(devID)` func so that it doesn't start a goroutine that just closes stuff